src/llmq/snapshot.cpp

```diff
@@ -288,55 +325,59 @@ bool BuildQuorumRotationInfo(CDeterministicMNManager& dmnman, CQuorumSnapshotMan
         }
 
         CSimplifiedMNListDiff mnhneeded;
-        if (!BuildSimplifiedMNListDiff(dmnman, chainman, qblockman, qman, GetLastBaseBlockHash(baseBlockIndexes, pNeededWorkBlockIndex), pNeededWorkBlockIndex->GetBlockHash(), mnhneeded, errorRet)) {
+        if (!BuildSimplifiedMNListDiff(dmnman, chainman, qblockman, qman, GetLastBaseBlockHash(baseBlockIndexes, pNeededWorkBlockIndex, use_legacy_construction), pNeededWorkBlockIndex->GetBlockHash(), mnhneeded, errorRet)) {
             return false;
         }
-        baseBlockIndexes.push_back(pNeededWorkBlockIndex);
-
+        if (!use_legacy_construction) {
+            baseBlockIndexes.push_back(pNeededWorkBlockIndex);
+        }
         response.mnListDiffList.push_back(mnhneeded);
     }
 
-    if (!BuildSimplifiedMNListDiff(dmnman, chainman, qblockman, qman,
-                                   GetLastBaseBlockHash(baseBlockIndexes, pWorkBlockHMinus3CIndex),
-                                   pWorkBlockHMinus3CIndex->GetBlockHash(), response.mnListDiffAtHMinus3C, errorRet)) {
-        return false;
-    }
-    baseBlockIndexes.push_back(pWorkBlockHMinus3CIndex);
+    if (!use_legacy_construction) {
```

Comment:
maybe for simplicity better to invert this condition? if legacy -> return true.

shorter diff, easier to read, and review.